Userland/Libraries/LibJS/Bytecode/Pass/AllocateRegisters.cpp

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2021, Peter Elliott <pelliott@ualberta.ca>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/NumericLimits.h>
+#include <AK/QuickSort.h>
+#include <LibJS/Bytecode/BasicBlock.h>
+#include <LibJS/Bytecode/PassManager.h>
+
+// This pass implements a simple linear-pass register allocator to reduce the
+// number of registers used in an executable, as described in the following paper:
+// Poletto, Massimiliano; Sarkar, Vivek (1999). "Linear scan register allocation".
+
+namespace JS::Bytecode::Passes {
+
+// This pass assumes that the registers are in single static Assingment form
```

Comment:
That's a pretty bold assumption
I _think_ we're still in some partial kind of SSA if we disregard the accumulator register, but you should probably not assume that without at least making sure that it's true (i.e. `VERIFY(is_in_ssa)`)